src/vsphere_cpi/spec/integration/ip_conflict_detection_spec.rb

```diff
@@ -56,4 +57,115 @@
       end
     end
   end
+
+  context 'when testing with fully qualified network names' do
```

Comment:
your context -> it statements you should be able to make a readable statement.

when a VM already exists on network when the VM exists on a different network with same unqualified name
may be it should be
when a VM exists with same IP and VM exists on a different network with same unqualified name

Reply:
you can use rspec -fd spec/ to see what your test statements look like